lib/Sema/CSApply.cpp

```diff
@@ -3333,7 +3333,9 @@ namespace {
 
               auto &tc = cs.getTypeChecker();
               tc.diagnose(cast->getLoc(), diag::conditional_downcast_foreign,
-                          destValueType);
+                          destValueType)
+                .highlight(cast->getSourceRange())
+                .fixItReplace(cast->getLoc(), "as!");
```

Comment:
Fix-Its should try to produce code that will compile, so the Fix-It here is technically incomplete because changing an `as?` to `as!` eliminates a layer of optional. For example, this would mean that code like:

```swift
if let cf = obj as? CFString { ... }
```

would become ill-formed if the Fix-It were accepted:

```swift
if let cf = obj as! CFString { ... }  // initializer for conditional binding must have Optional type, not `CFString`
```

Reply:
Opps. You are absolutely right.